File: Program Cpp/Garage.cpp
#pragma once

#include <iostream>
#include <string>
#include <vector>
#include "Vehicle.cpp"
#include "Car.cpp"
#include "Motorcycle.cpp"

using namespace std;

class Garage
{
	// Class Garage untuk parent dari class ParkingLot karena menurut saya itu saling berhubungan

private:
	string nama_garasi;
	string luas_garasi;
	vector<Car> daftar_kendaraan_mobil;
	vector<Motorcycle> daftar_kendaraan_motor;
	// Variable yang dibutuhkan
	// vector digunakan untuk melakukan array of object pada class Car dan Motorcycle
	// Melakukan Composite dari class Car dan Motorcycle yang dimasukan kedalam Vector

	
public:
	Garage(string nama_garasi, string luas_garasi){
		this->nama_garasi = nama_garasi;
		this->luas_garasi = luas_garasi;
	}
	// Constructor 


	Garage(string nama_garasi, string luas_garasi, Car daftar_kendaraan){
		this->nama_garasi = nama_garasi;
		this->luas_garasi = luas_garasi;
		this->daftar_kendaraan_mobil.push_back(daftar_kendaraan);
	}
	// Constructor: Overloading

	Garage(string nama_garasi, string luas_garasi, Motorcycle daftar_kendaraan){
		this->nama_garasi = nama_garasi;
		this->luas_garasi = luas_garasi;
		this->daftar_kendaraan_motor.push_back(daftar_kendaraan);
	}
	// Constructor: Overloading

	string getNamaGarasi(){
		return this->nama_garasi;
	}
	
	string getLuasGarasi(){
		return this->luas_garasi;
	}

	void addKendaraan(Car daftar_kendaraan){
		this->daftar_kendaraan_mobil.push_back(daftar_kendaraan);
	}

	void addKendaraan(Motorcycle daftar_kendaraan){
		this->daftar_kendaraan_motor.push_back(daftar_kendaraan);
	}
	// Function: Overloading

	vector<Car> getDaftarKendaraanMobil(){
		return this->daftar_kendaraan_mobil;
	}

	vector<Motorcycle> getDaftarKendaraanMotor(){
		return this->daftar_kendaraan_motor;
	}
};

File: Program Python/Garage.py
from Vehicle import Vehicle
from Car import Car
from Motorcycle import Motorcycle

class Garage:
    def __init__(self, name, size):
        self.name = name
        self.size = size
        self.cars = []
        self.motorcycles = []

    def add_vehicle(self, vehicle):
        if isinstance(vehicle, Car):
            self.cars.append(vehicle)
        elif isinstance(vehicle, Motorcycle):
            self.motorcycles.append(vehicle)
        else:
            raise ValueError("Invalid vehicle type")

    def get_daftar_kendaraan_mobil(self):
        return self.cars

    def get_daftar_kendaraan_motor(self):
        return self.motorcycles

    def get_nama_garasi(self):
        return self.name

    def get_luas_garasi(self):
        return self.size

    def __str__(self):
        return f"Garage: {self.name}, Size: {self.size}, Cars: {self.cars}, Motorcycles: {self.motorcycles}"